python/plugins/processing/algs/gdal/gdalcalc.py

```diff
@@ -105,24 +172,32 @@ def group(self):
     def groupId(self):
         return 'rastermiscellaneous'
 
+    def commandName(self):
+        if isWindows():
+            return 'gdal_calc'
+        else:
+            return 'gdal_calc.py'
+
     def getConsoleCommands(self, parameters, context, feedback, executing=True):
-        out = self.getOutputValue(self.OUTPUT)
-        extra = self.getParameterValue(self.EXTRA)
-        if extra is not None:
-            extra = str(extra)
-        #debug = self.getParameterValue(self.DEBUG)
-        formula = self.getParameterValue(self.FORMULA)
-        noData = self.getParameterValue(self.NO_DATA)
-        if noData is not None:
-            noData = str(noData)
+        if not parameters:
+            raise QgsProcessingException
+
+        out = self.parameterAsOutputLayer(parameters, self.OUTPUT, context)
+        extra = self.parameterAsString(parameters, self.EXTRA, context)
+        # if extra is not None:
+        #     extra = str(extra)
+        #debug = self.getParameterValue(parameters, self.DEBUG)
+        formula = self.parameterAsString(parameters, self.FORMULA, context)
+        noData = self.parameterAsDouble(parameters, self.NO_DATA, context)
```

Comment:
Here you need to handle noData = 0, so copy this logic:
https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/gdal/proximity.py#L145